File: GithubAnalyzer/Models/PullRequest.cs
namespace GithubAnalyzer.Models
{
    public class PullRequest
    {
        public int Number { get; set; }

        public string Title { get; set; }

        public string Body { get; set; }

        public int ChangedFiles { get; set; }

        public List<Commit> Commits { get; set; }

        public DateTime? CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }

        public DateTime? MergedAt { get; set; }

        public DateTime? ClosedAt { get; set; }

        public string State { get; set; } 

        public Author CreatedBy { get; set; }

        public List<Comment> Comments { get; set; }

        public List<Author> Assignees { get; set; }

        public List<Label> Labels { get; set; }

        public Author MergedBy { get; set; }

        public List<Review> Reviews { get; set; }

        public List<ReviewRequest> ReviewRequests { get; set; }
    }
}

File: github-miner/src/types/PullRequest.ts
import { PrBranch } from "./PrBranch";
import { Commit } from "./Commit";
import { Review } from "./Review";
import { ReviewRequest } from "./ReviewRequest";
import { User } from "./User";
import { Comment } from "./Comment";

export type PullRequest = {
  number?: number;
  title?: string;
  body?: string;
  head?: PrBranch;
  base?: PrBranch;
  changedFiles?: number;
  commits?: Commit[];
  createdAt?: string;
  updatedAt?: string;
  mergedAt?: string;
  closedAt?: string;
  state?: string;
  createdBy?: User;
  comments?: Comment[];
  assignees?: User[];
  labels?: string[];
  mergedBy?: User;
  reviews?: Review[];
  reviewRequests?: ReviewRequest[];
};
